packages/openapi3/test/test-servers.ts

```diff
@@ -0,0 +1,108 @@
+import { deepStrictEqual } from "assert";
+import { openApiFor } from "./test-host.js";
+
+describe("openapi3: servers", () => {
+  it("set a basic server", async () => {
+    const res = await openApiFor(
+      `
+      @serviceTitle("My service")
+      @server("https://example.com", "Main server")
+      namespace MyService {}
+      `
+    );
+    deepStrictEqual(res.servers, [
+      {
+        description: "Main server",
+        url: "https://example.com",
+        variables: {},
+      },
+    ]);
+  });
+
+  it("set a server with parameters", async () => {
+    const res = await openApiFor(
+      `
+      @serviceTitle("My service")
+      @server("https://{account}.{region}.example.com", "Regional account endpoint", {region: string, account: string})
```

Comment:
I think we should emit a warning diagnostic if there is no description for the server variables